examples/todomvc/src/scheduler.rs

```diff
@@ -57,46 +66,67 @@ impl Scheduler {
 
     /// Start the event loop, taking messages from the stack to run
     fn run(&self) {
-        let events_len = {
-            let Ok(events) = self.events.try_borrow() else { deadlock() };
-            events.len()
-        };
+        let mut events_len = 0;
+        {
+            if let Ok(events) = self.events.try_borrow() {
+                events_len = events.len();
+            } else {
+                exit("This might be a deadlock");
+            }
+        }
         if events_len == 0 {
-            let Ok(mut running) = self.running.try_borrow_mut() else { deadlock() };
-            *running = false;
+            if let Ok(mut running) = self.running.try_borrow_mut() {
+                *running = false;
+            } else {
+                exit("This might be a deadlock");
+            }
         } else {
             {
-                let Ok(mut running) = self.running.try_borrow_mut() else { deadlock() };
-                *running = true;
+                if let Ok(mut running) = self.running.try_borrow_mut() {
+                    *running = true;
+                } else {
+                    exit("This might be a deadlock");
+                }
             }
             self.next_message();
         }
     }
 
     fn next_message(&self) {
         let event = {
-            let Ok(mut events) = self.events.try_borrow_mut() else { deadlock() };
-            events.pop()
+            if let Ok(mut events) = self.events.try_borrow_mut() {
+                Some(events.pop())
+            } else {
+                exit("This might be a deadlock");
+                None
+            }
         };
-        if let Some(event) = event {
+        if let Some(Some(event)) = event {
             match event {
                 Message::Controller(e) => {
-                    let Ok(mut controller) = self.controller.try_borrow_mut() else { deadlock() };
-                    if let Some(ref mut ag) = *controller {
-                        ag.call(e);
+                    if let Ok(mut controller) = self.controller.try_borrow_mut() {
+                        if let Some(ref mut ag) = *controller {
+                            ag.call(e);
+                        }
+                    } else {
+                        exit("This might be a deadlock");
                     }
                 }
                 Message::View(e) => {
-                    let Ok(mut view) = self.view.try_borrow_mut() else { deadlock() };
-                    if let Some(ref mut ag) = *view {
-                        ag.call(e);
+                    if let Ok(mut view) = self.view.try_borrow_mut() {
+                        if let Some(ref mut ag) = *view {
+                            ag.call(e);
+                        }
+                    } else {
+                        exit("This might be a deadlock");
                     }
                 }
```

Comment:
As above.

Reply:
I don't see any `{ … }` that can be removed here.

Reply:
Me neither, my mistake.